aosp_diff/preliminary/external/drm_hwcomposer/0006-fix-crtc-connector-static-bound-to-dynamaically-bound.patch

```diff
@@ -0,0 +1,237 @@
+From 2192343c4196bceb8f03a9f48ca75609357636ad Mon Sep 17 00:00:00 2001
+From: Xiao HaowenX <haowenx.xiao@intel.com>
+Date: Tue, 9 Nov 2021 14:55:59 +0800
+Subject: [PATCH] fix crtc connector static bound to dynamaically bound
+
+Tracked-On:OAM-99492
+Signed-off-by: Xiao HaowenX <haowenx.xiao@intel.com>
+---
+ DrmHwcTwo.cpp         | 73 ++++++++++++++++++++++++++++++++++++++++---
+ DrmHwcTwo.h           |  2 ++
+ drm/DrmDevice.cpp     | 41 +++---------------------
+ drm/DrmDevice.h       |  5 ++-
+ drm/ResourceManager.h |  3 ++
+ 5 files changed, 82 insertions(+), 42 deletions(-)
+
+diff --git a/DrmHwcTwo.cpp b/DrmHwcTwo.cpp
+index 194b11e..fe806a7 100644
+--- a/DrmHwcTwo.cpp
++++ b/DrmHwcTwo.cpp
+@@ -1202,23 +1202,83 @@ void DrmHwcTwo::HandleInitialHotplugState(DrmDevice *drmDevice) {
+   }
+ }
+ 
++int DrmHwcTwo::AttachWriteback_(DrmConnector *display_conn,DrmDevice *drm) {
++  DrmCrtc *display_crtc = display_conn->encoder()->crtc();
++  if (drm->GetWritebackConnectorForDisplay(display_crtc->display()) != nullptr) {
++    ALOGE("Display already has writeback attach to it");
++    return -EINVAL;
++  }
++  for (auto &writeback_conn : drm->writeback_connectors()) {
++    if (writeback_conn->display() >= 0)
++      continue;
++    for (DrmEncoder *writeback_enc : writeback_conn->possible_encoders()) {
++      for (DrmCrtc *possible_crtc : writeback_enc->possible_crtcs()) {
++        if (possible_crtc != display_crtc)
++          continue;
++        // Use just encoders which had not been bound already
++        if (writeback_enc->can_bind(display_crtc->display())) {
++          writeback_enc->set_crtc(display_crtc);
++          writeback_conn->set_encoder(writeback_enc);
++          writeback_conn->set_display(display_crtc->display());
++          writeback_conn->UpdateModes();
++          return 0;
++        }
++      }
++    }
++  }
++  return -EINVAL;
++}
++int DrmHwcTwo::DynamaicallyBound(DrmConnector *mconn,DrmDevice *mdrm,DrmHwcTwo *mhwc2){
++	int display_id;
++	int curr_num_display = mhwc2->resource_manager_.getDisplayCount();
++	mconn->set_display(curr_num_display);
++	display_id = mconn->display();
++	for (DrmEncoder *enc : mconn->possible_encoders()) {
```

Comment:
recommand to directly call DrmDevice::CreateDisplayPipe to bind, can avoid the duplication and be more reliable .